src/features/wallet/root/ActivityCard/ActivityCardHeader.tsx

```diff
@@ -1,34 +1,39 @@
-import React, { memo, useCallback, useState } from 'react'
+import React, { memo, useCallback, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
 import CardHandle from '../../../../components/CardHandle'
 import Box from '../../../../components/Box'
 import { FilterKeys, FilterType } from '../walletTypes'
 import ModalPicker from '../../../../components/ModalPicker'
 import { useAppDispatch } from '../../../../store/store'
 import activitySlice from '../../../../store/activity/activitySlice'
-import animateTransition from '../../../../utils/animateTransition'
 import accountSlice from '../../../../store/account/accountSlice'
 
 type Props = { filter: FilterType }
 const ActivityCardHeader = ({ filter }: Props) => {
   const { t } = useTranslation()
-  const filters = t('transactions.filter', { returnObjects: true }) as Record<
-    string,
-    string
-  >
+  const filters = useMemo(
+    () =>
+      t('transactions.filter', { returnObjects: true }) as Record<
+        string,
+        string
+      >,
+    [t],
+  )
   const dispatch = useAppDispatch()
-  const [data] = useState(
-    FilterKeys.map((value) => ({ label: filters[value], value })),
+  const data = useMemo(
+    () => FilterKeys.map((value) => ({ label: filters[value], value })),
+    [filters],
   )
 
   const onFilterChanged = useCallback(
     (_itemValue, itemIndex) => {
       const nextFilter = FilterKeys[itemIndex]
-      animateTransition()
-      dispatch(accountSlice.actions.resetActivityChart())
-      dispatch(activitySlice.actions.setFilter(nextFilter))
+      if (nextFilter !== filter) {
```

Comment:
I'm curious if this was related to the language change or just a general optimization?

Reply:
Just a general optimization, but when you change the language this gets triggered a couple times so figured it was worth it to check

Reply:
The animation I removed does seem to cause bugs though, I sometimes see a overlap of the graphs etc so I removed it for now

Reply:
ok, sounds good.